superset-frontend/src/explore/components/controls/FilterControl/AdhocFilterEditPopoverSimpleTabContent/index.jsx

```diff
@@ -236,16 +237,39 @@ export default class AdhocFilterEditPopoverSimpleTabContent extends React.Compon
   }
 
   isOperatorRelevant(operator, subject) {
+    const column = this.props.datasource.columns?.find(
+      col => col.column_name === subject,
+    );
+    const isColumnBoolean =
+      !!column && (column.type === 'BOOL' || column.type === 'BOOLEAN');
+    const isColumnNumber = !!column && column.type === 'INT';
+    const isColumnFunction = !!column && !!column.expression;
+
     if (operator && CUSTOM_OPERATORS.has(operator)) {
       const { partitionColumn } = this.props;
       return partitionColumn && subject && subject === partitionColumn;
     }
-
+    if (
+      operator === OPERATORS['IS TRUE'] ||
+      operator === OPERATORS['IS FALSE']
+    ) {
+      return isColumnBoolean || isColumnNumber || isColumnFunction;
+    }
+    if (isColumnBoolean) {
+      return (
+        operator === OPERATORS['IS NULL'] ||
+        operator === OPERATORS['IS NOT NULL']
+      );
+    }
     return !(
       (this.props.datasource.type === 'druid' &&
-        TABLE_ONLY_OPERATORS.indexOf(operator) >= 0) ||
+        [...BOOLEAN_ONLY_OPERATORS, ...TABLE_ONLY_OPERATORS].indexOf(
+          operator,
+        ) >= 0) ||
       (this.props.datasource.type === 'table' &&
-        DRUID_ONLY_OPERATORS.indexOf(operator) >= 0) ||
+        [...BOOLEAN_ONLY_OPERATORS, ...DRUID_ONLY_OPERATORS].indexOf(
+          operator,
+        ) >= 0) ||
```

Comment:
You can revert changes in these two lines, too, since you've checked boolean operators above.